src/main/java/org/mule/extension/aggregator/internal/config/AggregatorManager.java

```diff
@@ -188,6 +188,23 @@ public void registerListener(String aggregatorName, AggregatorListener listener)
     registeredListeners.put(aggregatorName, listener);
   }
 
+  /**
+   * Unregisters a listener from an already registered aggregator
+   *
+   * @param aggregatorName the name of the aggregator to unregister from
+   * @param listener the listener to be called when needed
+   * @throws MuleRuntimeException
+   */
+  public void unregisterListener(String aggregatorName, AggregatorListener listener) throws MuleRuntimeException {
+    if (!availableAggregators.containsKey(aggregatorName)) {
+      throw new MuleRuntimeException(createStaticMessage("Listener is attempting to unregister from aggregator: '%s', but it does not exist",
+                                                         aggregatorName));
+    }
+    if (registeredListeners.containsKey(aggregatorName)) {
```

Comment:
This line is not needed. According to [this](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#remove-java.lang.Object-java.lang.Object-), Map.remove(key,value) is equivalente to:
```
if (map.containsKey(key) && Objects.equals(map.get(key), value)) {
     map.remove(key);
     return true;
 } else
     return false;
```